### ISSUE TITLE
Rule checking for async or await variable names reports false positives

### ISSUE THREAD

**Issue by tvsonar** [SonarSource/sonarlint-vs#284]
Reported by a user:
```
ConfiguredTaskAwaitable myMethod = AsyncMethod().ConfigureAwait();
await myMethod;
```
